Topics/python-fundamentals.md

```diff
@@ -0,0 +1,236 @@
+
+From [this article](https://www.freecodecamp.org/news/what-is-git-and-how-to-use-it-c341b049ae61/):
+
+- Control System: This basically means that Git is a content tracker. So Git can be used to store content — it is mostly used to store code due to the other features it provides.
+
+- Version Control System: The code which is stored in Git keeps changing as more code is added. Also, many developers can add code in parallel. So Version Control System helps in handling this by maintaining a history of what changes have happened. Also, Git provides features like branches and merges, which I will be covering later.
+
+- Distributed Version Control System: Git has a remote repository that is stored in a server and a local repository that is stored in the computer of each developer. This means that the code is not just stored in a central server, but the full copy of the code is present in all the developers’ computers. Git is a Distributed Version Control System since the code is present in every developer’s computer. I will explain the concept of remote and local repositories later in this article.
+
+_Any existing software project that is not under version control is considered a dead project, and the responsible developers are considered crazy._
+
+Data Science projects (which make heavy use of software) are no different, indeed! 
+
+They also have the _additional problem of data versioning_, which is the raw material on which you work most. 
+
+[**Here**](https://rogerdudler.github.io/git-guide/index.html) you can find a simple guide to Git.
+
+**Learn it, it's freaking worth (and necessary).**
+
+Documenting well your work with Git is crucial: read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
+**Must read:**
+[**Ten Simple Rules for Taking Advantage of Git and GitHub**](http://dx.plos.org/10.1371/journal.pcbi.1004947)
+
+### Ask Questions
+A rule of thumb to learn fast and effectively is to ask questions and read other's questions and answers.
+
+Join communities of people interested in the topic (e.g. Reddit): here you can find discussions, search by keywords (e.g. "matrix multiplication"), and ask questions, with experts who will answer and help you.
+
+Some _tips_ regarding questions:
+
+- Try to form specific, well-written questions, to minimize the time used by the respondent.
+- Do not ask a question whose answer is found with a quick search on google.
+- If the questions are too general or show laziness they'll likely remain unanswered...
+
+Some subreddits you can subscribe to are:
+
+- [r/Python](https://www.reddit.com/r/python/)
+- [r/LearnPython](https://www.reddit.com/r/learnpython/)
+- [r/LearnProgramming](https://www.reddit.com/r/learnprogramming/)
+- [r/Programming](https://www.reddit.com/r/programming/)
+- [r/Computer Science](https://www.reddit.com/r/compsci/)
+
+Other good places to post and read (well structured) questions are:
```

Comment:
Take a look at [this link](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/things-to-know.md#things-to-know) and see if you can use some of the details from there or backlink it here

Reply:
k